openfga/v1/openfga_service.proto

```diff
@@ -1175,6 +1243,66 @@ message CheckResponse {
   string resolution = 2;
 }
 
+message BatchCheckRequest {
+  string store_id = 1 [
+    json_name = "store_id",
+    (google.api.field_behavior) = REQUIRED,
+    (validate.rules).string = {pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01YCP46JKYM8FJCQ37NMBYHE5X\""}
+  ];
+
+  repeated BatchCheckItem checks = 2 [(google.api.field_behavior) = REQUIRED];
+
+  string authorization_model_id = 3 [
+    json_name = "authorization_model_id",
+    (validate.rules).string = {
+      pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""}
+  ];
+
+  ConsistencyPreference consistency = 4 [(validate.rules).enum.defined_only = true];
+}
+
+message BatchCheckItem {
+  CheckRequestTupleKey tuple_key = 1 [(google.api.field_behavior) = REQUIRED];
+  openfga.v1.ContextualTupleKeys contextual_tuples = 2 [json_name = "contextual_tuples"];
+  google.protobuf.Struct context = 3;
+
+  string correlation_id = 4 [
+    (validate.rules).string = {pattern: "^[\\w\\d-]{1,36}$"},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1cd93d8c-8e45-43c6-9a15-cbb3c7f394bc\""
+      description: "correlation_id must be a string containing only letters, numbers, or hyphens, with length ≤ 36 characters."
+    },
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+message BatchCheckResponse {
+  map<string, BatchCheckSingleResult> result = 1 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    example: '{"1cd93d8c-8e45-43c6-9a15-cbb3c7f394bc": {"allowed": true, "queryDurationMs": 3, "error": {"message": ""}}}'
+    description: "map keys are the correlation_id values from the BatchCheckItems in the request"
+  }];
+}
+
+message BatchCheckSingleResult {
+  oneof check_result {
+    bool allowed = 2;
```

Comment:
why are we starting from 2?

Reply:
ah good eye, definitely a leftover from earlier iterations